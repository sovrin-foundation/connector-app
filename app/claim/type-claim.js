// @flow

import type {
  GenericObject,
  CustomError,
  InitialTestAction,
  ResetAction,
  NotificationPayload,
} from '../common/type-common'
import type { ClaimPushPayload } from '../push-notification/type-push-notification'

export type Claim = {
  messageId: string,
  claim: { [string]: Array<string> },
  schema_seq_no: number,
  issuer_did: string,
  signature: {
    primary_claim: {
      m2: string,
      a: string,
      e: string,
      v: string,
    },
    non_revocation_claim?: GenericObject,
  },
  optional_data?: GenericObject,
  remoteDid: string,
  uid: string,
  from_did: string,
  forDID: string,
  connectionHandle?: number,
  remotePairwiseDID?: string,
}

export type ClaimWithUuid = Claim & {
  claim_uuid: string,
}

export const CLAIM_RECEIVED = 'CLAIM_RECEIVED'
export type ClaimReceivedAction = {
  type: typeof CLAIM_RECEIVED,
  claim: Claim,
}

export const CLAIM_STORAGE_SUCCESS = 'CLAIM_STORAGE_SUCCESS'
export type ClaimStorageSuccessAction = {
  type: typeof CLAIM_STORAGE_SUCCESS,
  messageId: string,
}

export const CLAIM_STORAGE_FAIL = 'CLAIM_STORAGE_FAIL'
export type ClaimStorageFailAction = {
  type: typeof CLAIM_STORAGE_FAIL,
  messageId: string,
  error: CustomError,
}

export const MAP_CLAIM_TO_SENDER = 'MAP_CLAIM_TO_SENDER'
export type MapClaimToSenderAction = {
  type: typeof MAP_CLAIM_TO_SENDER,
  claimUuid: string,
  senderDID: string,
  myPairwiseDID: string,
  logoUrl: string,
}

export type ClaimMap = {
  +[claimUuid: string]: {
    senderDID: string,
    myPairwiseDID: string,
    logoUrl: string,
  },
}

export const HYDRATE_CLAIM_MAP = 'HYDRATE_CLAIM_MAP'
export type HydrateClaimMapAction = {
  type: typeof HYDRATE_CLAIM_MAP,
  claimMap: ClaimMap,
}

export const HYDRATE_CLAIM_MAP_FAIL = 'HYDRATE_CLAIM_MAP_FAIL'
export type HydrateClaimMapFailAction = {
  type: typeof HYDRATE_CLAIM_MAP_FAIL,
  error: CustomError,
}

export type ClaimPushPayloadVcx = {
  connectionHandle: number,
}
export type ClaimVcx = NotificationPayload & ClaimPushPayloadVcx

export const CLAIM_RECEIVED_VCX = 'CLAIM_RECEIVED_VCX'
export type ClaimReceivedVcxAction = {
  type: typeof HYDRATE_CLAIM_MAP_FAIL,
  claim: ClaimVcx,
}

export type ClaimAction =
  | ClaimReceivedAction
  | ClaimStorageSuccessAction
  | ClaimStorageFailAction
  | MapClaimToSenderAction
  | HydrateClaimMapAction
  | HydrateClaimMapFailAction
  | InitialTestAction
  | ResetAction
  | ClaimReceivedVcxAction

export type ClaimStore = {
  +[string]: {
    claim: Claim,
    error?: CustomError,
  },
  claimMap: ClaimMap,
}

export const ERROR_CLAIM_HYDRATE_FAIL = {
  message: 'Failed to hydrate claim map',
  code: 'CL-001',
}

export type GetClaimVcxResult = {
  claimUuid: string,
  claim: ClaimPushPayload,
}
