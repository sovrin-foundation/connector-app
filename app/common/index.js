// @flow
export * from './button-constants'
export * from './message-constants'
export * from './route-constants'
export * from './secure-storage-constants'
export function noop() {}
