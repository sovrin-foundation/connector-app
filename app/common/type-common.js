// @flow

import type { Element } from 'react'
import type { NavigationScreenProp } from 'react-navigation'

export type CustomError = {
  code: string,
  message: string,
}

export const INITIAL_TEST_ACTION: 'INITIAL_TEST_ACTION' = 'INITIAL_TEST_ACTION'

export type InitialTestAction = {
  type: typeof INITIAL_TEST_ACTION,
}

export const initialTestAction = () => ({
  type: INITIAL_TEST_ACTION,
})

export type NavigationParams = {
  [string]: any,
}
export type NavigationRoute = NavigationLeafRoute
export type NavigationState = {
  index: number,
  routes: Array<NavigationRoute>,
}

export type NavigationLeafRoute = {|
  key: string,
  routeName: string,
  path?: string,
  params?: NavigationParams,
|}

export type ReactNavigation = NavigationScreenProp

export type GenericObject = {
  [string]: any,
}

export type GenericStringObject = {
  [string]: string,
}

export type MessageAnnotation = {
  name: string,
  version: string,
}

export type TopicAnnotation = {
  tid: number,
  mid: number,
}

export type ReactChildren = Element<*>

export const RESET: 'RESET' = 'RESET'
export const REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS: 'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS' =
  'REMOVE_SERIALIZED_CLAIM_OFFERS_SUCCESS'

export type ResetAction = {
  type: typeof RESET,
}

export type ImageSource = {
  uri: string,
}

export type NotificationPayload = {
  forDID: string,
  uid: string,
  type: string,
  remotePairwiseDID: string,
  senderLogoUrl?: ?string,
}

export type ImmutableGenericStringObject = {
  +[string]: string,
}

export const STORE_STATUS = {
  IDLE: 'IDLE',
  IN_PROGRESS: 'IN_PROGRESS',
  ERROR: 'ERROR',
  SUCCESS: 'SUCCESS',
}

export type StoreError = { +error: ?CustomError }
export type StoreStatus = { +status: $Keys<typeof STORE_STATUS> }

export type StatusBarStyle = 'default' | 'light-content' | 'dark-content'
