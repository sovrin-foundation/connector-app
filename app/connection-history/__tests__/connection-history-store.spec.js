// @flow
import {
  all,
  takeLatest,
  takeEvery,
  put,
  call,
  select,
} from 'redux-saga/effects'
import connectionHistoryReducer, {
  loadHistory,
  loadHistorySuccess,
  loadHistoryFail,
  loadHistorySaga,
  recordHistoryEvent,
  historyEventOccurred,
  historyEventOccurredSaga,
  convertInvitationToHistoryEvent,
  convertConnectionSuccessToHistoryEvent,
  convertClaimStorageSuccessToHistoryEvent,
  convertProofRequestToHistoryEvent,
  convertProofSendToHistoryEvent,
  convertSendClaimRequestSuccessToHistoryEvent,
  deleteHistoryEvent,
} from '../connection-history-store'
import { initialTestAction } from '../../common/type-common'
import {
  getTestInvitationPayload,
  successConnectionData,
  claimOffer,
  claimRequest,
  claim,
  claimOfferPayload,
  pendingClaimHistory,
  proofRequest,
  senderDid1,
  fulfilledRequestedAttributes,
  invitationReceivedEvent,
  newConnectionSuccessEvent,
  sendClaimRequestSuccessEvent,
  claimReceivedEvent,
  claimReceivedSuccessEvent,
  proofRequestReceivedEvent,
  proofRequestAutofillEvent,
  proofSharedEvent,
  proofRequestAutofill,
  proof,
  uid,
} from '../../../__mocks__/static-data'
import { invitationRejected } from '../../invitation/invitation-store'
import { saveNewConnection } from '../../store/connections-store'
import {
  claimOfferReceived,
  claimRequestSuccess,
  sendClaimRequestSuccess,
} from '../../claim-offer/claim-offer-store'
import { claimReceived, claimStorageSuccess } from '../../claim/claim-store'
import { CLAIM_STORAGE_SUCCESS } from '../../claim/type-claim'
import {
  proofRequestReceived,
  sendProof,
  proofRequestAutoFill,
  sendProofSuccess,
} from '../../proof-request/proof-request-store'
import {
  HISTORY_EVENT_STORAGE_KEY,
  ERROR_LOADING_HISTORY,
  ERROR_HISTORY_EVENT_OCCURRED,
} from '../type-connection-history'
import {
  getProofRequest,
  getProof,
  getClaimOffer,
  getPendingHistoryEvent,
  getHistoryEvent,
  getPendingHistory,
  getClaimReceivedHistory,
} from '../../store/store-selector'
import { getHydrationItem } from '../../services/storage'
import {
  CLAIM_OFFER_RECEIVED,
  SEND_CLAIM_REQUEST_SUCCESS,
} from './../../claim-offer/type-claim-offer'
import { RESET } from '../../common/type-common'
import { PROOF_REQUEST_RECEIVED } from '../../proof-request/type-proof-request'

jest.mock('../../services/uuid')

function getHistoryData() {
  // generate history data from static data that we have

  const invitationGenerator = getTestInvitationPayload()
  const invitationPayload = invitationGenerator.next().value
  let sender1History = []
  if (invitationPayload) {
    // add history for connection request
    sender1History.push(
      convertInvitationToHistoryEvent(invitationPayload.payload)
    )
  }

  // add history for connection success
  sender1History.push(
    convertConnectionSuccessToHistoryEvent(
      saveNewConnection(successConnectionData)
    )
  )
  sender1History.push(
    convertSendClaimRequestSuccessToHistoryEvent(
      sendClaimRequestSuccess(uid, claimOfferPayload)
    )
  )

  // add history for proof request
  sender1History.push(
    convertProofRequestToHistoryEvent(
      proofRequestReceived(proofRequest.payload, proofRequest.payloadInfo)
    )
  )

  // add history for proof sent
  sender1History.push(
    convertProofSendToHistoryEvent(
      proofSharedEvent,
      proofRequestAutofill,
      proof
    )
  )

  return {
    [senderDid1]: sender1History,
  }
}

describe('Store: ConnectionHistory', () => {
  let initialState
  beforeEach(() => {
    initialState = connectionHistoryReducer(undefined, initialTestAction())
  })

  it('match loadHistory', () => {
    expect(
      connectionHistoryReducer(initialState, loadHistory())
    ).toMatchSnapshot()
  })

  it('match loadHistorySuccess', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        loadHistorySuccess(getHistoryData())
      )
    ).toMatchSnapshot()
  })

  it('match loadHistoryFail', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        loadHistoryFail({
          code: 'TEST 101',
          message: 'Load History failed',
        })
      )
    ).toMatchSnapshot()
  })

  it('match record history event', () => {
    expect(
      connectionHistoryReducer(
        initialState,
        recordHistoryEvent(
          convertSendClaimRequestSuccessToHistoryEvent(
            sendClaimRequestSuccess(uid, claimOfferPayload)
          )
        )
      )
    ).toMatchSnapshot()
  })

  it('historyEventOccurredSaga should raise success for correct invitation received ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(invitationReceivedEvent)
    )
    historyEvent = convertInvitationToHistoryEvent(
      invitationReceivedEvent.data.payload
    )

    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for correct new connection ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(newConnectionSuccessEvent)
    )
    historyEvent = convertConnectionSuccessToHistoryEvent(
      newConnectionSuccessEvent
    )

    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for sending claim request ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(sendClaimRequestSuccessEvent)
    )
    historyEvent = convertSendClaimRequestSuccessToHistoryEvent(
      sendClaimRequestSuccessEvent
    )
    expect(gen.next().value).toEqual(
      select(
        getHistoryEvent,
        historyEvent.originalPayload.uid,
        historyEvent.remoteDid,
        CLAIM_OFFER_RECEIVED
      )
    )
    expect(gen.next().value).toEqual(
      select(
        getPendingHistory,
        historyEvent.originalPayload.uid,
        historyEvent.remoteDid,
        SEND_CLAIM_REQUEST_SUCCESS
      )
    )

    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for claim received ', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(claimReceivedSuccessEvent)
    )

    expect(gen.next().value).toEqual(
      select(getClaimOffer, claimReceivedSuccessEvent.messageId)
    )

    historyEvent = convertClaimStorageSuccessToHistoryEvent(
      claimReceivedSuccessEvent,
      claimOfferPayload
    )
    expect(gen.next(claimOfferPayload).value).toEqual(
      select(
        getClaimReceivedHistory,
        historyEvent.originalPayload.messageId,
        historyEvent.remoteDid,
        CLAIM_STORAGE_SUCCESS
      )
    )

    expect(gen.next().value).toEqual(
      select(getPendingHistoryEvent, claimOfferPayload)
    )
    expect(gen.next(pendingClaimHistory).value).toEqual(
      put(deleteHistoryEvent(pendingClaimHistory))
    )
  })

  it('historyEventOccurredSaga should raise success for correct proof request received', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(
      historyEventOccurred(proofRequestReceivedEvent)
    )
    historyEvent = convertProofRequestToHistoryEvent(proofRequestReceivedEvent)
    expect(gen.next().value).toEqual(
      select(
        getHistoryEvent,
        historyEvent.originalPayload.payloadInfo.uid,
        historyEvent.remoteDid,
        PROOF_REQUEST_RECEIVED
      )
    )
    expect(gen.next().value).toEqual(put(recordHistoryEvent(historyEvent)))
  })

  it('historyEventOccurredSaga should raise success for proof shared', () => {
    let historyEvent
    const gen = historyEventOccurredSaga(historyEventOccurred(proofSharedEvent))
    expect(gen.next().value).toEqual(
      select(getProofRequest, proofSharedEvent.uid)
    )
    expect(gen.next(proofRequestAutofill).value).toEqual(
      select(getProof, proofSharedEvent.uid)
    )
    historyEvent = convertProofSendToHistoryEvent(
      proofSharedEvent,
      proofRequestAutofill,
      proof
    )
  })

  it('historyEventOccurredSaga should raise failure in case anything fails', () => {
    const gen = historyEventOccurredSaga(historyEventOccurred(proofSharedEvent))
    expect(gen.next().value).toEqual(
      select(getProofRequest, proofSharedEvent.uid)
    )
    const error = new Error()
    expect(gen.throw(error).value).toEqual(
      put(
        loadHistoryFail({
          ...ERROR_HISTORY_EVENT_OCCURRED,
          message: `${ERROR_HISTORY_EVENT_OCCURRED.message} ${error.message}`,
        })
      )
    )
  })

  it('convertInvitationToHistoryEvent should raise success', () => {
    expect(
      convertInvitationToHistoryEvent(invitationReceivedEvent.data.payload)
    ).toMatchSnapshot()
  })

  it('convertConnectionSuccessToHistoryEvent should raise success', () => {
    expect(
      convertConnectionSuccessToHistoryEvent(newConnectionSuccessEvent)
    ).toMatchSnapshot()
  })

  it('convertSendClaimRequestSuccessToHistoryEvent should raise success', () => {
    expect(
      convertSendClaimRequestSuccessToHistoryEvent(sendClaimRequestSuccessEvent)
    ).toMatchSnapshot()
  })

  it('convertProofRequestToHistoryEvent should raise success', () => {
    expect(
      convertProofRequestToHistoryEvent(proofRequestReceivedEvent)
    ).toMatchSnapshot()
  })

  it('convertProofAutoFillToHistoryEvent should raise success', () => {
    expect(
      convertProofSendToHistoryEvent(
        proofSharedEvent,
        proofRequestAutofill,
        proof
      )
    ).toMatchSnapshot()
  })

  it('should raise failure in case data getHydrationItem fails', () => {
    const gen = loadHistorySaga()
    gen.next()
    expect(gen.next().value).toEqual(
      call(getHydrationItem, HISTORY_EVENT_STORAGE_KEY)
    )
    const error = new Error()
    expect(gen.throw(error).value).toEqual(
      put(
        loadHistoryFail({
          ...ERROR_LOADING_HISTORY,
          message: `${ERROR_LOADING_HISTORY.message} ${error.message}`,
        })
      )
    )
    expect(gen.next().done).toBe(true)
  })

  it('should reset history if RESET action is raised', () => {
    const afterOneHistoryEventState = connectionHistoryReducer(
      initialState,
      recordHistoryEvent(
        convertSendClaimRequestSuccessToHistoryEvent(
          sendClaimRequestSuccess(uid, claimOfferPayload)
        )
      )
    )
    expect(
      connectionHistoryReducer(afterOneHistoryEventState, {
        type: RESET,
      })
    ).toMatchSnapshot()
  })
})
