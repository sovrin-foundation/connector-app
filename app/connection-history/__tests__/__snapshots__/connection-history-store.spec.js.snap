// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Store: ConnectionHistory convertConnectionSuccessToHistoryEvent should raise success 1`] = `
Object {
  "action": "CONNECTED",
  "data": Array [
    Object {
      "data": "2018-01-10T04:32:43+05:30",
      "label": "Established on",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "sender1",
  "originalPayload": Object {
    "connection": Object {
      "identifier": "pairwiseIdentifier1",
      "logoUrl": "lu",
      "myPairwiseAgentDid": "myPairwiseAgentDID",
      "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
      "myPairwiseDid": "pairwiseIdentifier1",
      "myPairwisePeerVerKey": "senderVerificationKey1",
      "myPairwiseVerKey": "pairwiseVerificationKey1",
      "requestId": "requestId1",
      "senderAgencyDetail": Object {
        "DID": "senderDID1",
        "endpoint": "endpoint",
        "verKey": "senderVerificationKey1",
      },
      "senderAgentKeyDelegationProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "senderDID": "senderDID1",
      "senderDetail": Object {
        "DID": "senderDID1",
        "agentKeyDlgProof": Object {
          "agentDID": "senderAgentDID1",
          "agentDelegatedKey": "agentDelegatedKey",
          "signature": "signature",
        },
        "logoUrl": "http://testissuer.com/logoUrl.png",
        "name": "sender1",
        "verKey": "senderVerificationKey1",
      },
      "senderEndpoint": "endpoint",
      "senderLogoUrl": "lu",
      "senderName": "sender1",
      "senderVerificationKey": "sVk",
      "targetName": "target name",
    },
    "type": "NEW_CONNECTION_SUCCESS",
  },
  "remoteDid": "senderDID1",
  "status": "CONNECTED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "INVITATION",
}
`;

exports[`Store: ConnectionHistory convertInvitationToHistoryEvent should raise success 1`] = `
Object {
  "action": "CONNECTION REQUEST",
  "data": Object {},
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "sender1",
  "originalPayload": Object {
    "requestId": "requestId1",
    "senderAgencyDetail": Object {
      "DID": "senderDID1",
      "endpoint": "endpoint",
      "verKey": "senderVerificationKey1",
    },
    "senderAgentKeyDelegationProof": Object {
      "agentDID": "senderAgentDID1",
      "agentDelegatedKey": "agentDelegatedKey",
      "signature": "signature",
    },
    "senderDID": "senderDID1",
    "senderDetail": Object {
      "DID": "senderDID1",
      "agentKeyDlgProof": Object {
        "agentDID": "senderAgentDID1",
        "agentDelegatedKey": "agentDelegatedKey",
        "signature": "signature",
      },
      "logoUrl": "http://testissuer.com/logoUrl.png",
      "name": "sender1",
      "verKey": "senderVerificationKey1",
    },
    "senderEndpoint": "endpoint",
    "senderLogoUrl": "lu",
    "senderName": "sender1",
    "senderVerificationKey": "sVk",
    "targetName": "target name",
  },
  "remoteDid": "senderDID1",
  "status": "CONNECTION REQUEST",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "INVITATION",
}
`;

exports[`Store: ConnectionHistory convertProofAutoFillToHistoryEvent should raise success 1`] = `
Object {
  "action": "SHARED",
  "data": Array [
    Object {
      "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
      "data": "male",
      "key": "attr2_uuid",
      "label": "Address 2",
    },
    Object {
      "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
      "data": "Alex",
      "key": "attr1_uuid",
      "label": "Address 1",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "proof_req_auto_filled",
  "originalPayload": Object {
    "type": "SEND_PROOF_SUCCESS",
    "uid": "pid123",
  },
  "remoteDid": "remotePairWiseDID",
  "status": "SHARED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "PROOF",
}
`;

exports[`Store: ConnectionHistory convertProofRequestToHistoryEvent should raise success 1`] = `
Object {
  "action": "PROOF RECEIVED",
  "data": Array [
    Object {
      "label": "Address 1",
    },
    Object {
      "label": "Address 2",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "Home Address",
  "originalPayload": Object {
    "payload": Object {
      "data": Object {
        "name": "Home Address",
        "requestedAttributes": Array [
          Object {
            "label": "Address 1",
          },
          Object {
            "label": "Address 2",
          },
        ],
        "version": "1.0.0",
      },
      "originalProofRequestData": Object {
        "name": "proof_req_1",
        "nonce": "123432421212",
        "requested_attributes": Object {
          "attr1_uuid": Object {
            "name": "Address 1",
          },
          "attr2_uuid": Object {
            "name": "Address 2",
          },
        },
        "requested_predicates": Object {},
        "version": "0.1",
      },
      "proofHandle": 1,
      "requester": Object {
        "name": "Test Issuer",
      },
      "statusMsg": "pending",
    },
    "payloadInfo": Object {
      "remotePairwiseDID": "senderDID1",
      "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
      "uid": "pid123",
    },
    "type": "PROOF_REQUEST_RECEIVED",
  },
  "remoteDid": "senderDID1",
  "status": "PROOF RECEIVED",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "PROOF",
}
`;

exports[`Store: ConnectionHistory convertSendClaimRequestSuccessToHistoryEvent should raise success 1`] = `
Object {
  "action": "PENDING",
  "data": Array [
    Object {
      "data": "Address Address Address",
      "label": "Address 1",
    },
    Object {
      "data": "Address 2 Address 2 Address 2",
      "label": "Address 2",
    },
  ],
  "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
  "name": "Home Address",
  "originalPayload": Object {
    "payload": Object {
      "claimRequestStatus": "CLAIM_REQUEST_SUCCESS",
      "data": Object {
        "claimDefinitionSchemaSequenceNumber": 36,
        "name": "Home Address",
        "revealedAttributes": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "version": "1.0.0",
      },
      "issuer": Object {
        "did": "issuerDid",
        "name": "Test Issuer",
      },
      "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
      "senderLogoUrl": "http://testissuer.com/logoUrl.png",
      "status": "RECEIVED",
      "statusMsg": "pending",
      "uid": "usd123",
    },
    "type": "SEND_CLAIM_REQUEST_SUCCESS",
    "uid": "uid",
  },
  "remoteDid": "ha66899sadfjZJGINKN0770",
  "status": "PENDING",
  "timestamp": "2018-01-10T04:32:43+05:30",
  "type": "CLAIM",
}
`;

exports[`Store: ConnectionHistory match loadHistory 1`] = `
Object {
  "data": null,
  "error": null,
  "isLoading": true,
}
`;

exports[`Store: ConnectionHistory match loadHistoryFail 1`] = `
Object {
  "data": null,
  "error": Object {
    "code": "TEST 101",
    "message": "Load History failed",
  },
  "isLoading": false,
}
`;

exports[`Store: ConnectionHistory match loadHistorySuccess 1`] = `
Object {
  "data": Object {
    "senderDID1": Array [
      Object {
        "action": "CONNECTION REQUEST",
        "data": Object {},
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "sender1",
        "originalPayload": Object {
          "requestId": "requestId1",
          "senderAgencyDetail": Object {
            "DID": "senderDID1",
            "endpoint": "endpoint",
            "verKey": "senderVerificationKey1",
          },
          "senderAgentKeyDelegationProof": Object {
            "agentDID": "senderAgentDID1",
            "agentDelegatedKey": "agentDelegatedKey",
            "signature": "signature",
          },
          "senderDID": "senderDID1",
          "senderDetail": Object {
            "DID": "senderDID1",
            "agentKeyDlgProof": Object {
              "agentDID": "senderAgentDID1",
              "agentDelegatedKey": "agentDelegatedKey",
              "signature": "signature",
            },
            "logoUrl": "http://testissuer.com/logoUrl.png",
            "name": "sender1",
            "verKey": "senderVerificationKey1",
          },
          "senderEndpoint": "endpoint",
          "senderLogoUrl": "lu",
          "senderName": "sender1",
          "senderVerificationKey": "sVk",
          "targetName": "target name",
        },
        "remoteDid": "senderDID1",
        "status": "CONNECTION REQUEST",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "INVITATION",
      },
      Object {
        "action": "CONNECTED",
        "data": Array [
          Object {
            "data": "2018-01-10T04:32:43+05:30",
            "label": "Established on",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": undefined,
        "originalPayload": Object {
          "connection": Object {
            "newConnection": Object {
              "identifier": "pairwiseIdentifier1",
              "logoUrl": "lu",
              "myPairwiseAgentDid": "myPairwiseAgentDID",
              "myPairwiseAgentVerKey": "myPairwiseAgentVerKey",
              "myPairwiseDid": "pairwiseIdentifier1",
              "myPairwisePeerVerKey": "senderVerificationKey1",
              "myPairwiseVerKey": "pairwiseVerificationKey1",
              "requestId": "requestId1",
              "senderAgencyDetail": Object {
                "DID": "senderDID1",
                "endpoint": "endpoint",
                "verKey": "senderVerificationKey1",
              },
              "senderAgentKeyDelegationProof": Object {
                "agentDID": "senderAgentDID1",
                "agentDelegatedKey": "agentDelegatedKey",
                "signature": "signature",
              },
              "senderDID": "senderDID1",
              "senderDetail": Object {
                "DID": "senderDID1",
                "agentKeyDlgProof": Object {
                  "agentDID": "senderAgentDID1",
                  "agentDelegatedKey": "agentDelegatedKey",
                  "signature": "signature",
                },
                "logoUrl": "http://testissuer.com/logoUrl.png",
                "name": "sender1",
                "verKey": "senderVerificationKey1",
              },
              "senderEndpoint": "endpoint",
              "senderLogoUrl": "lu",
              "senderName": "sender1",
              "senderVerificationKey": "sVk",
              "targetName": "target name",
            },
          },
          "type": "NEW_CONNECTION",
        },
        "remoteDid": undefined,
        "status": "CONNECTED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "INVITATION",
      },
      Object {
        "action": "PENDING",
        "data": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "claimRequestStatus": "CLAIM_REQUEST_SUCCESS",
            "data": Object {
              "claimDefinitionSchemaSequenceNumber": 36,
              "name": "Home Address",
              "revealedAttributes": Array [
                Object {
                  "data": "Address Address Address",
                  "label": "Address 1",
                },
                Object {
                  "data": "Address 2 Address 2 Address 2",
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "issuer": Object {
              "did": "issuerDid",
              "name": "Test Issuer",
            },
            "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
            "senderLogoUrl": "http://testissuer.com/logoUrl.png",
            "status": "RECEIVED",
            "statusMsg": "pending",
            "uid": "usd123",
          },
          "type": "SEND_CLAIM_REQUEST_SUCCESS",
          "uid": "uid",
        },
        "remoteDid": "ha66899sadfjZJGINKN0770",
        "status": "PENDING",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "CLAIM",
      },
      Object {
        "action": "PROOF RECEIVED",
        "data": Array [
          Object {
            "label": "Address 1",
          },
          Object {
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "data": Object {
              "name": "Home Address",
              "requestedAttributes": Array [
                Object {
                  "label": "Address 1",
                },
                Object {
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "originalProofRequestData": Object {
              "name": "proof_req_1",
              "nonce": "123432421212",
              "requested_attributes": Object {
                "attr1_uuid": Object {
                  "name": "Address 1",
                },
                "attr2_uuid": Object {
                  "name": "Address 2",
                },
              },
              "requested_predicates": Object {},
              "version": "0.1",
            },
            "proofHandle": 1,
            "requester": Object {
              "name": "Test Issuer",
            },
            "statusMsg": "pending",
          },
          "payloadInfo": Object {
            "remotePairwiseDID": "senderDID1",
            "senderLogoUrl": "http://cr0ybot.github.io/ingress-logos/ingress.png",
            "uid": "pid123",
          },
          "type": "PROOF_REQUEST_RECEIVED",
        },
        "remoteDid": "senderDID1",
        "status": "PROOF RECEIVED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "PROOF",
      },
      Object {
        "action": "SHARED",
        "data": Array [
          Object {
            "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
            "data": "male",
            "key": "attr2_uuid",
            "label": "Address 2",
          },
          Object {
            "claimUuid": "claim::f760213b-e719-47fb-9669-b72c5c410e8c",
            "data": "Alex",
            "key": "attr1_uuid",
            "label": "Address 1",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "proof_req_auto_filled",
        "originalPayload": Object {
          "type": "SEND_PROOF_SUCCESS",
          "uid": "pid123",
        },
        "remoteDid": "remotePairWiseDID",
        "status": "SHARED",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "PROOF",
      },
    ],
  },
  "error": null,
  "isLoading": false,
}
`;

exports[`Store: ConnectionHistory match record history event 1`] = `
Object {
  "data": Object {
    "ha66899sadfjZJGINKN0770": Array [
      Object {
        "action": "PENDING",
        "data": Array [
          Object {
            "data": "Address Address Address",
            "label": "Address 1",
          },
          Object {
            "data": "Address 2 Address 2 Address 2",
            "label": "Address 2",
          },
        ],
        "id": "a4f35623-b50c-40ea-a2b0-f7cd06e03142",
        "name": "Home Address",
        "originalPayload": Object {
          "payload": Object {
            "claimRequestStatus": "CLAIM_REQUEST_SUCCESS",
            "data": Object {
              "claimDefinitionSchemaSequenceNumber": 36,
              "name": "Home Address",
              "revealedAttributes": Array [
                Object {
                  "data": "Address Address Address",
                  "label": "Address 1",
                },
                Object {
                  "data": "Address 2 Address 2 Address 2",
                  "label": "Address 2",
                },
              ],
              "version": "1.0.0",
            },
            "issuer": Object {
              "did": "issuerDid",
              "name": "Test Issuer",
            },
            "remotePairwiseDID": "ha66899sadfjZJGINKN0770",
            "senderLogoUrl": "http://testissuer.com/logoUrl.png",
            "status": "RECEIVED",
            "statusMsg": "pending",
            "uid": "usd123",
          },
          "type": "SEND_CLAIM_REQUEST_SUCCESS",
          "uid": "uid",
        },
        "remoteDid": "ha66899sadfjZJGINKN0770",
        "status": "PENDING",
        "timestamp": "2018-01-10T04:32:43+05:30",
        "type": "CLAIM",
      },
    ],
  },
  "error": null,
  "isLoading": false,
}
`;

exports[`Store: ConnectionHistory should reset history if RESET action is raised 1`] = `
Object {
  "data": null,
  "error": null,
  "isLoading": false,
}
`;
